File: 2026/EventPlanner/backend/src/main/java/com/event/service/EventService.java
package com.event.service;

import com.event.model.Event;
import com.event.repository.EventRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class EventService {

    private final EventRepository repository;

    public EventService(EventRepository repository) {
        this.repository = repository;
    }

    public List<Event> getAll() {
        return repository.findAll();
    }

    public Event getById(Long id) {
        return repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Event not found with id " + id));
    }

    public List<Event> getByCategory(String category) {
        if (category == null || category.isBlank()) return getAll();
        return repository.findByCategoryIgnoreCase(category.trim());
    }

    public List<Event> getByStatus(String status) {
        if (status == null || status.isBlank()) return getAll();
        return repository.findByStatusIgnoreCase(status.trim());
    }

    public List<Event> getByDateRange(LocalDateTime start, LocalDateTime end) {
        if (start == null || end == null) return getAll();
        return repository.findByStartDateTimeBetween(start, end);
    }

    public List<Event> search(String term) {
        if (term == null || term.isBlank()) return getAll();
        return repository.findByTitleContainingIgnoreCase(term.trim());
    }

    public Event create(Event event) {
        return repository.save(event);
    }

    public Event update(Long id, Event data) {
        Event existing = getById(id);
        existing.setTitle(data.getTitle());
        existing.setDescription(data.getDescription());
        existing.setLocation(data.getLocation());
        existing.setStartDateTime(data.getStartDateTime());
        existing.setEndDateTime(data.getEndDateTime());
        existing.setCategory(data.getCategory());
        existing.setStatus(data.getStatus());
        return repository.save(existing);
    }

    public void delete(Long id) {
        Event existing = getById(id);
        repository.delete(existing);
    }
}

File: 2026/EventPlanner/frontend/src/app/services/event.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { EventItem } from '../models/event.model';

@Injectable({
  providedIn: 'root'
})
export class EventService {
  private apiUrl = 'http://localhost:8094/api/events';

  constructor(private http: HttpClient) {}

  getAll(): Observable<EventItem[]> {
    return this.http.get<EventItem[]>(this.apiUrl);
  }

  getById(id: number): Observable<EventItem> {
    return this.http.get<EventItem>(`${this.apiUrl}/${id}`);
  }

  getByCategory(category: string): Observable<EventItem[]> {
    return this.http.get<EventItem[]>(`${this.apiUrl}/category/${encodeURIComponent(category)}`);
  }

  getByStatus(status: string): Observable<EventItem[]> {
    return this.http.get<EventItem[]>(`${this.apiUrl}/status/${encodeURIComponent(status)}`);
  }

  getByRange(start: string, end: string): Observable<EventItem[]> {
    return this.http.get<EventItem[]>(`${this.apiUrl}/range?start=${encodeURIComponent(start)}&end=${encodeURIComponent(end)}`);
  }

  search(q: string): Observable<EventItem[]> {
    return this.http.get<EventItem[]>(`${this.apiUrl}/search?q=${encodeURIComponent(q)}`);
  }

  create(event: EventItem): Observable<EventItem> {
    return this.http.post<EventItem>(this.apiUrl, event);
  }

  update(id: number, event: EventItem): Observable<EventItem> {
    return this.http.put<EventItem>(`${this.apiUrl}/${id}`, event);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
